_posts/2019-03-06-lecture-15.md

```diff
@@ -0,0 +1,256 @@
+
+
+
+Thus, we see that the graphs in a graphical model represent the model itself whereas it just represents computation in the case of a deep network.
+
+Some neural networks can be completely understood as a conventional graphical model. A few examples are examined below - 
+## 1.Restricted Boltzmann Machines
+
+<figure id="RBM">
+      <img src="{{ 'assets/img/notes/lecture-15/RBM.png' | relative_url }}" />
+</figure>
+
+**Overview**: RBMs are represented by undirected bipartite graphs, with one "visible" layer a second "hidden" layer. There are no connections between nodes in the same layer (in contrast to *unrestricted* Boltzmann machines, which permit connections between hidden units).
+
+**Learning**: We compute a loss function by marginalizing over all the hidden variables and taking the derivative w.r.t weights:
+
+
+
+$$
+\log L(v) = \log \sum_{h} \exp(\sum_{i,j} w_{ij} v_{i} h_{j} + \sum_{i} b_{i}v_{i} + \sum_{j} c_{j} h_{j} - \log(Z))
+$$
+
+$$
+\frac{\partial \log L(v)}{\partial w_{ij}} = \sum_{h} P(h | v) \frac{\partial}{\partial w_{ij}} P(v,h) - \sum_{v,h} P(v,h) \frac{\partial}{\partial w_{ij}} P(v,h)
+$$
+
+or alternatively, as: 
+$$
+\dfrac{\partial}{\partial w_{ij}} \log L(v) = \mathbb{E}_{P(h|v)} \bigg[\dfrac{\partial}{\partial w_{ij}} P(v, h)\bigg] - \mathbb{E}_{P(v, h)} \bigg[\dfrac{\partial}{\partial w_{ij}} P(v, h)\bigg]
+$$
+
+Of the two terms in the loss above:
+
+- The first term is the expectation taking over the conditional of the hidden given the observed, ie. we *average over the posterior*. This is computed via sampling, which is exact (the RBM factorizes over $h$ given $v$).  In neural network literature, this is also called **the clamped**/ **wake**/ **positive phase**.
+- The second term is the expectation taken over all random variables (both hidden and observed), i.e we *average over the joint*. This is also computed sampling, but requires the iterative Gibbs sampling algorithm. Consequently, it's far more expensive (computationally) and slow to converge. In neural network literature, this is called the **the unclamped**/ **sleep**/ **negative phase**.
+
+## 2. Sigmoid Belief Networks
+
+SBNs consist of stacks of directed linear layers combined via non-linear sigmoid activation functions. The intermediate meaning of the hidden layers are typically ignored and so even though they synactically represent graphical models, operationally they begin to deviate. Operationally treating them as graphical models, leads to inefficient inference because of the implicit "explaining away" phenomenon as when we want to infer a hidden variable, it is coupled with all other hidden random variables in that layer. This stands in contrast to the RMB case because *d*-seperation in the RBM graph meant that the hidden random variables are independent given the other layer. Also, unlike the RBM, computing the conditional does not involve a negative phase.
```

Comment:
```suggestion
SBNs consist of stacks of directed linear layers combined via non-linear sigmoid activation functions. The intermediate meaning of the hidden layers are typically ignored and so even though they synactically represent graphical models, operationally they begin to deviate. Operationally treating them as graphical models, leads to inefficient inference because of the implicit "explaining away" phenomenon as when we want to infer a hidden variable, it is coupled with all other hidden random variables in that layer. This stands in contrast to the RBM case because *d*-seperation in the RBM graph meant that the hidden random variables are independent given the other layer. Also, unlike the RBM, computing the conditional does not involve a negative phase.
```